src/leaderboard/messages/data/fetch_by_score.rs

```diff
@@ -0,0 +1,184 @@
+use super::{fetch::FetchResponse, Order, RankedElement};
+use crate::leaderboard::MomentoRequest;
+use crate::utils::prep_leaderboard_request_with_timeout;
+use crate::{Leaderboard, MomentoError, MomentoErrorCode, MomentoResult};
+
+use momento_protos::common::Unbounded;
+use momento_protos::leaderboard::score_range::{Max, Min};
+
+use std::ops::Range;
+
+/// Represents a range of scores used to request elements by score.
+pub struct ScoreRange {
+    min: Option<f64>,
+    max: Option<f64>,
+}
+
+impl ScoreRange {
+    /// Constructs a new `ScoreRange`.
+    pub fn new(min: Option<f64>, max: Option<f64>) -> Self {
+        Self { min, max }
+    }
+
+    /// Constructs a new `ScoreRange` with no bounds.
+    pub fn unbounded() -> Self {
+        Self {
+            min: None,
+            max: None,
+        }
+    }
+
+    /// Validates the score range.
+    pub fn validate(&self) -> MomentoResult<()> {
+        if let Some(min) = self.min {
+            if !min.is_finite() && min != f64::NEG_INFINITY {
+                return Err(MomentoError {
+                    message: format!("min score must be finite or negative infinity; got {}", min),
+                    error_code: MomentoErrorCode::InvalidArgumentError,
+                    inner_error: None,
+                    details: None,
+                });
+            }
+        }
+        if let Some(max) = self.max {
+            if !max.is_finite() && max != f64::INFINITY {
+                return Err(MomentoError {
+                    message: format!("max score must be finite or positive infinity; got {}", max),
+                    error_code: MomentoErrorCode::InvalidArgumentError,
+                    inner_error: None,
+                    details: None,
+                });
+            }
+        }
+        Ok(())
+    }
+}
+
+impl From<Option<ScoreRange>> for ScoreRange {
+    fn from(val: Option<ScoreRange>) -> Self {
+        val.unwrap_or_else(ScoreRange::unbounded)
+    }
+}
```

Comment:
This appears to be unused, and it is unclear to me that unbounded means the same thing as None for a whole score range. If you want to keep it, what's the utility here, and how should this be communicated so it doesn't surprise people that `None` means `Everything`?

Reply:
Good catch -- this was a leftover while I was iterating on the UI. I've removed this in [e930128](https://github.com/momentohq/client-sdk-rust/pull/438/commits/e93012847c0f00f949768003d48bb0eb4815a07c)